internal/driver/network.go

```diff
@@ -271,29 +266,30 @@ func (d NetworkDriver) DeleteNetwork(request *network.DeleteNetworkRequest) erro
 	return nil
 }
 
-func (d NetworkDriver) findDockerContainerByEndpointId(endpointID string) (containerAndIPAddress, error) {
+func (d NetworkDriver) findDockerContainerByEndpointId(endpointID string) (dockerTypes.Container, string, error) {
 	containers, err := d.dockerCli.ContainerList(context.Background(), dockerTypes.ContainerListOptions{})
 	if err != nil {
 		err = errors.Wrap(err, "dockerCli ContainerList Error")
 		log.Errorln(err)
-		return containerAndIPAddress{}, err
+		return dockerTypes.Container{}, "", err
 	}
 	for _, container := range containers {
 		for _, network := range container.NetworkSettings.Networks {
 			if endpointID == network.EndpointID {
-				return containerAndIPAddress{container, network.IPAddress}, nil
+				return container, network.IPAddress, nil
 			}
 		}
 	}
-	return containerAndIPAddress{}, errors.Errorf("find no container with endpintID = %s", endpointID)
+	return dockerTypes.Container{}, "", errors.Errorf("find no container with endpintID = %s", endpointID)
 }
 
-func (containerAndIP containerAndIPAddress) reserveOrReleaseIPByFixedIPLabel(ripam ReservedIPManager) error {
-	_, fixedIP := containerAndIP.container.Labels[fixedIPLabel]
-	if fixedIP {
-		return ripam.MarkReservedIP(context.Background(), containerAndIP.ipAddress)
+func reserveIPIfContainerHasFixedIPLabel(ripam ReservedIPManager, container dockerTypes.Container, address string) {
```

Comment:
in my opinions ```IfXXX``` as a part of function name is a little bit strange, usually, it indicates ```if``` checking should be put into upper layer, like
```
if container.HasFixedIPLabel {
    reserveIP(...)
}
```

Reply:
ok, I will split the procedure into two method calls